nodestream_plugin_neptune/neptune_connection.py

```diff
@@ -0,0 +1,115 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+
+from aiobotocore.session import get_session
+
+
+class NeptuneConnection(ABC):
+    async def execute(self, query_stmt: str, parameters):
+        response = None
+        async with self._create_boto_client() as client:
+            try:
+                response = await self._execute_query(client,
+                    query_stmt=query_stmt,
+                    parameters=parameters,
+                )
+
+                code = response["ResponseMetadata"]["HTTPStatusCode"]
+                if code != 200:
+                    self.logger.error(
+                        f"Query `{query_stmt}` failed with response:\n{response}"
+                    )
+
+            except Exception as e:
+                self.logger.exception(
+                    f"Unexpected error for query: {query_stmt}.", e, stack_info=True
+                )
+
+        return response
+
+    @abstractmethod
+    def _create_boto_client(self):
+        pass
+
+    @abstractmethod
+    async def _execute_query(self, client, query_stmt: str, parameters: str):
+        pass
+
+
+class NeptuneDBConnection(NeptuneConnection):
+    @classmethod
+    def from_configuration(cls, host: str, region: str, graph_id: str = None):
+        if host is None or region is None:
+            raise ValueError(
+                "A `host` and `region` must be specified when `mode` is 'database'."
+            )
+        if graph_id is not None:
+            raise ValueError(
+                "A `graph_id` should not be used with Neptune Database, `host=<Neptune Endpoint>` should be used instead. If using Neptune Analytics, set `mode='analytics'."
+            )
+        return cls(host=host, region=region)
+
+    def __init__(self, host: str, region: str) -> None:
+        self.host = host
+        self.region = region
+        self.boto_session = get_session()
+        self.logger = getLogger(self.__class__.__name__)
+
+    def _create_boto_client(self):
+        return self.boto_session.create_client(
+            "neptunedata", region_name=self.region, endpoint_url=self.host
+        )
+
+    async def _execute_query(self, client, query_stmt: str, parameters):
+        self.logger.debug(
+            "Executing Cypher Query to Neptune Database",
+            extra={
+                "query": query_stmt,
+                "host": self.host,
+            },
+        )
+
+        return await client.execute_open_cypher_query(
+            openCypherQuery=query_stmt,
+            # Use json.dumps() to warp dict's key/values in double quotes.
+            parameters=json.dumps(parameters),
+        )
+
+
+class NeptuneAnalyticsConnection(NeptuneConnection):
```

Comment:
nit: Adding class documentation, explaining the connection to Neptune vs Analytics and links to the DBs.